From afc82d91805c93de2fb93dbc47128c818e2fd947 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 3 Jun 2024 16:30:20 +0200
Subject: IWYU: returning an std::unique_ptr requires WebMediaPlayer type
 definition

Fixed missing include of WebMediaPlayer in WebLocalFrameClient.

But doing just that breaks root_scroller_test.cc in MacOS, that will
bring that dependency and include a macro fixed1 from Carbon. So
change the name of the variable to avoid the clash.

Bug: 41455655
Change-Id: I3b398f56783a4d273f54059b19fb7794c27b42db
---
 .../blink/public/web/web_local_frame_client.h        |  1 +
 .../core/page/scrolling/root_scroller_test.cc        | 12 ++++++------
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/third_party/blink/public/web/web_local_frame_client.h b/third_party/blink/public/web/web_local_frame_client.h
index 03bf5ef3fc666..1eafc236ca40b 100644
--- a/third_party/blink/public/web/web_local_frame_client.h
+++ b/third_party/blink/public/web/web_local_frame_client.h
@@ -74,6 +74,7 @@
 #include "third_party/blink/public/platform/web_content_settings_client.h"
 #include "third_party/blink/public/platform/web_effective_connection_type.h"
 #include "third_party/blink/public/platform/web_file_system_type.h"
+#include "third_party/blink/public/platform/web_media_player.h"
 #include "third_party/blink/public/platform/web_prescient_networking.h"
 #include "third_party/blink/public/platform/web_set_sink_id_callbacks.h"
 #include "third_party/blink/public/platform/web_source_location.h"
diff --git a/third_party/blink/renderer/core/page/scrolling/root_scroller_test.cc b/third_party/blink/renderer/core/page/scrolling/root_scroller_test.cc
index 7f10326106490..668c991b84693 100644
--- a/third_party/blink/renderer/core/page/scrolling/root_scroller_test.cc
+++ b/third_party/blink/renderer/core/page/scrolling/root_scroller_test.cc
@@ -2373,22 +2373,22 @@ TEST_F(ImplicitRootScrollerSimTest, BottomFixedAffectedByTopControls) {
       To<HTMLFrameOwnerElement>(container1)->contentDocument();
   Document* child2_document =
       To<HTMLFrameOwnerElement>(container2)->contentDocument();
-  LayoutObject* fixed =
+  LayoutObject* fixed_layout =
       GetDocument().getElementById(AtomicString("fixed"))->GetLayoutObject();
-  LayoutObject* fixed1 =
+  LayoutObject* fixed_layout1 =
       child1_document->getElementById(AtomicString("fixed"))->GetLayoutObject();
-  LayoutObject* fixed2 =
+  LayoutObject* fixed_layout2 =
       child2_document->getElementById(AtomicString("fixed"))->GetLayoutObject();
 
-  EXPECT_TRUE(fixed->FirstFragment()
+  EXPECT_TRUE(fixed_layout->FirstFragment()
                   .PaintProperties()
                   ->PaintOffsetTranslation()
                   ->IsAffectedByOuterViewportBoundsDelta());
-  EXPECT_TRUE(fixed1->FirstFragment()
+  EXPECT_TRUE(fixed_layout1->FirstFragment()
                   .PaintProperties()
                   ->PaintOffsetTranslation()
                   ->IsAffectedByOuterViewportBoundsDelta());
-  EXPECT_FALSE(fixed2->FirstFragment()
+  EXPECT_FALSE(fixed_layout2->FirstFragment()
                    .PaintProperties()
                    ->PaintOffsetTranslation()
                    ->IsAffectedByOuterViewportBoundsDelta());
-- 
2.43.0

